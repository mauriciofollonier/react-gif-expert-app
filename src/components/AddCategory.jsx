import { useState } from "react";
import PropTypes from "prop-types";


// props
// 1).- setCategories recibia la func desde el componente padre para poder cambiar el estado
// aca en este componente.
// 2).- onNewCategory envia un evento al componente padre con el nuevo valor (inputValue).
//                                     1)                2)
export const AddCategory = ( { /* setCategories */ onNewCategory } ) => {

  const [inputValue, setInputValue] = useState('');  

  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  }

  const onSubmit = ( event ) => {
    event.preventDefault();
    if( inputValue.trim().length <= 1) return; // Si los datos ingresados son menor
                                               // o igual a un caracter, que no lo mande
    // 1)                                      
    // setCategories( (categories) => [ inputValue, ...categories ] );
    // 2)
    onNewCategory( inputValue.trim() );

    setInputValue(''); // Una vez agregada la categoria limpiamos el input.
  }


  return (
    <form onSubmit={ onSubmit } aria-label='form'>
        <input 
         type="text"
         placeholder="Buscar gifs" 
         value = { inputValue }
         onChange={ onInputChange }
        />
    </form>
  )
}

AddCategory.propTypes = {
  
  onNewCategory: PropTypes.func.isRequired 
}