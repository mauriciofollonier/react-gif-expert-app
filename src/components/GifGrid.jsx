import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {

  // Custom Hoook
     const { images, isLoading } = useFetchGifs( category );

  /* TODA ESTA LOGICA PASO AL CUSTOM HOOK - useFetchGifs() */
  
  // const [images, setImages] = useState([]);
  // const getImages = async () => {
  // const newImages = await getGifs( category );
  // setImages( newImages );
  // }
  // useEffect( () => {
  //   getImages();
  // }, [ ])

  return (
    <>
        <h3>{ category }</h3>

        {
          isLoading && <h2>Cargando...</h2>
        }

        <div className='card-grid'>
           {
            images.map((image)=> (
                <GifItem 
                  key = { image.id }
                  { ...image } // Pasamos todo el objeto con todas las props 
                  />
              ))
           }
        </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}