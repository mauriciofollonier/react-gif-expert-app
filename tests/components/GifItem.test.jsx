import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";



describe('Pruebas en componente <GifItem/>', () => { 

    const title = 'Aegon Targaryen';
    const url = 'https://www.gameofthrones.com/jonsnow';

    test('Debe hacer match con el snapshot', () => { 

        const { container } = render(<GifItem title = {title} url = { url }/>);
        expect( container ).toMatchSnapshot(); 
     });

     test('Debe mostrar la imagen con el URL y ALT indicado', () => { 

        render(<GifItem title = {title} url = { url }/>);
        // Formas de evaluar 
        // 1).-
        // expect( screen.getByRole('img').src ).toBe( url );
        // expect( screen.getByRole('img').alt ).toBe( title );

        // 2).-
        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( title );
     });

     test('Debe mostrar el titulo', () => { 
        
        render(<GifItem title = {title} url = { url }/>);
        expect( screen.getByText( title )).toBeTruthy();    
      });
 })