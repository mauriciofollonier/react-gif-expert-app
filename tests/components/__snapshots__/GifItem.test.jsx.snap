// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <GifItem/> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Aegon Targaryen"
      src="https://www.gameofthrones.com/jonsnow"
    />
    <p>
      Aegon Targaryen
    </p>
  </div>
</div>
`;
